### PR TITLE
feat(quantic): change sortCriteria for facetValue

### PR DESCRIPTION
**Pull request by MSLI-Coveo**
[SVCC-2293](https://coveord.atlassian.net/browse/SVCC-2293)
change sortCriteria to "score" for facetValue in order to keep the same order of facetValues after "Show More" or "Show Less" is clicked.

### CODE REVIEW

**Comment by nathanlb** [comment]
The other possible values for sortCriteria should work as well though. Changing the default value doesn't feel like a real fix given then users that explicitly set it to something else would still have the same issue.

**Comment by erocheleau** [comment]
> The other possible values for sortCriteria should work as well though. Changing the default value doesn't feel like a real fix given users that explicitly set it to something else would still have the same issue.

The problem is that we were sending: `automatic` as the sortCriteria which is not an actually real value for the sortCriterias.

And the search API has a little bit of a weird default value for facet sort: 

> By default: 
When isFieldExpanded is false in the facet request, and moreValuesAvailable is true in the corresponding facet response, use score
Otherwise, use alphanumeric.

So by sending a value that doesn't exist, effectively when pressing show more, it would actually change our sorting criteria...

**Comment by nathanlb** [comment]
> > The other possible values for sortCriteria should work as well though. Changing the default value doesn't feel like a real fix given users that explicitly set it to something else would still have the same issue.
> 
> The problem is that we were sending: `automatic` as the sortCriteria which is not an actually real value for the sortCriterias.
> 
> And the search API has a little bit of a weird default value for facet sort:
> 
> > By default:
> > When isFieldExpanded is false in the facet request, and moreValuesAvailable is true in the corresponding facet response, use score
> > Otherwise, use alphanumeric.
> 
> So by sending a value that doesn't exist, effectively when pressing show more, it would actually change our sorting criteria from `score` to `alphanumeric`

Ahh je comprend. Yeah I recently realized that it wasn't a supported value and brought it up to the search ui team. They had no idea and had been using automatic for a long time. They decided to keep it as a way to explicitly say "not specified". This change makes sense then but we might as well remove any mention of `'automatic'` from the JSDoc then.

**Comment by MSLI-Coveo** [comment]
> > > The other possible values for sortCriteria should work as well though. Changing the default value doesn't feel like a real fix given users that explicitly set it to something else would still have the same issue.
> > 
> > 
> > The problem is that we were sending: `automatic` as the sortCriteria which is not an actually real value for the sortCriterias.
> > And the search API has a little bit of a weird default value for facet sort:
> > > By default:
> > > When isFieldExpanded is false in the facet request, and moreValuesAvailable is true in the corresponding facet response, use score
> > > Otherwise, use alphanumeric.
> > 
> > 
> > So by sending a value that doesn't exist, effectively when pressing show more, it would actually change our sorting criteria from `score` to `alphanumeric`
> 
> Ahh je comprend. Yeah I recently realized that it wasn't a supported value and brought it up to the search ui team. They had no idea and had been using automatic for a long time. They decided to keep it as a way to explicitly say "not specified". This change makes sense then but we might as well remove any mention of `'automatic'` from the JSDoc while we're at it.

And setting `sortCriteria` to `alphanumeric` would also trigger untracked change of content in some scenarios. It seems we still to use aria-live utility to announce that facetValues are change. Like the example below:

chrome-capture-2022-8-1 (2)

File: packages/quantic/force-app/main/default/lwc/quanticFacet/quanticFacet.js [hunk 1]
```diff
@@ -79,7 +79,7 @@ export default class QuanticFacet extends LightningElement {
    * @type  {'score' | 'alphanumeric' | 'occurrences' | 'automatic'}
    * @defaultValue `'automatic'`
```
Comment (nathanlb) [inline packages/quantic/force-app/main/default/lwc/quanticFacet/quanticFacet.js:80]:
```suggestion
   * @defaultValue `'alphanumeric'`
```

File: packages/quantic/force-app/main/default/lwc/quanticFacet/quanticFacet.js [hunk 2]
```diff
@@ -79,7 +79,7 @@ export default class QuanticFacet extends LightningElement {
    * @type  {'score' | 'alphanumeric' | 'occurrences' | 'automatic'}
```
Comment (nathanlb) [inline packages/quantic/force-app/main/default/lwc/quanticFacet/quanticFacet.js:79]:
```suggestion
   * @type  {'score' | 'alphanumeric' | 'occurrences'}
```

Comment (MSLI-Coveo) [reply packages/quantic/force-app/main/default/lwc/quanticFacet/quanticFacet.js:79]: 'automatic' removed from faceValue's sortCriteria type.

File: packages/quantic/force-app/main/default/lwc/quanticFacet/quanticFacet.js (hunk 1 above)
Comment (MSLI-Coveo) [reply packages/quantic/force-app/main/default/lwc/quanticFacet/quanticFacet.js]: change default value for sortCriteria to 'alphanumeric'

File: packages/quantic/force-app/main/default/lwc/quanticFacet/quanticFacet.js [hunk 3]
```diff
@@ -76,10 +76,10 @@ export default class QuanticFacet extends LightningElement {
    *   - `occurrences`
    *   - `automatic`
    * @api
-   * @type  {'score' | 'alphanumeric' | 'occurrences' | 'automatic'}
-   * @defaultValue `'automatic'`
+   * @type  {'score' | 'alphanumeric' | 'occurrences'}
+   * @defaultValue `'alphanumeric'`
```
Comment (erocheleau) [inline packages/quantic/force-app/main/default/lwc/quanticFacet/quanticFacet.js:80]: We should use 'score' as a default instead of alphanumeric though.

**Review by erocheleau — CHANGES_REQUESTED** [review]

File: packages/quantic/force-app/main/default/lwc/quanticFacet/quanticFacet.js (hunk 3 above)
Comment (MSLI-Coveo) [reply packages/quantic/force-app/main/default/lwc/quanticFacet/quanticFacet.js]: Changed @api sortCriteria to score for FacetValue

**Review by erocheleau — APPROVED** [review]

**Pull request merged**